src/modules/mavlink/mavlink_mission.cpp

```diff
@@ -691,6 +691,12 @@ MavlinkMissionManager::handle_mission_request_list(const mavlink_message_t *msg)
 	mavlink_msg_mission_request_list_decode(msg, &wprl);
 
 	if (CHECK_SYSID_COMPID_MISSION(wprl)) {
+		const bool maybe_completed = (_transfer_seq == current_item_count());
+
+		if (maybe_completed) {
+			switch_to_idle_state();
```

Comment:
I infer from this call and your narration that you're ignoring the last ACK by the ground station because with the current implementation it does not add any benefit to not just be able to restart a mission transmission in case that ACK was lost.

Reply:
I think you have some faulty assumptions on the protocol:
https://mavlink.io/en/services/mission.html#download_mission

Mission items are not acked, only the full mission after the last mission item has been received.

If the mission ACK gets lost, QGC won't know about it and happily trott along to request the next mission. But the state machine here remains stuck in the wrong state, waiting for the MISSION_ACK. But since single mission items are not acked, only re-requested if not received, we can assume that the last one was received properly (most likely at least), if a new mission request list comes in